packages/contracts/test/voting/admin.ts

```diff
@@ -0,0 +1,220 @@
+        ERRORS.ALREADY_INITIALIZED
+      );
+    });
+  });
+
+  describe('plugin interface: ', async () => {
+    it('supports plugin cloneable interface', async () => {
+      // @ts-ignore
+      const IPlugin = await hre.artifacts.readArtifact('IPlugin');
+
+      const iface = new ethers.utils.Interface(IPlugin.abi);
+
+      expect(await plugin.supportsInterface(getInterfaceID(iface))).to.be.eq(
+        true
+      );
+    });
+
+    it('supports admin address plugin interface', async () => {
+      const iface = new ethers.utils.Interface([
+        'function initialize(address  _dao)',
+        'function executeProposal(bytes _proposalMetadata, tuple(address,uint256,bytes)[] _actions)',
+      ]);
+
+      expect(await plugin.supportsInterface(getInterfaceID(iface))).to.be.eq(
+        true
+      );
+    });
+  });
+
+  describe('execute proposal: ', async () => {
+    beforeEach(async () => {
+      await initializePlugin();
+    });
+
+    it("fails to call DAO's `execute()` if `EXECUTE_PERMISSION` is not granted to the plugin address", async () => {
+      await dao.revoke(dao.address, plugin.address, EXECUTE_PERMISSION_ID);
+
+      await expect(
+        plugin.executeProposal(dummyMetadata, dummyActions)
+      ).to.be.revertedWith(
+        customError(
+          'Unauthorized',
+          dao.address,
+          dao.address,
+          plugin.address,
+          EXECUTE_PERMISSION_ID
+        )
+      );
+    });
+
+    it('fails to call `executeProposal()` if `ADMIN_EXECUTE_PERMISSION` is not granted for the admin address', async () => {
+      await dao.revoke(
+        plugin.address,
+        ownerAddress,
+        EXECUTE_PROPOSAL_PERMISSION_ID
+      );
+
+      await expect(
+        plugin.executeProposal(dummyMetadata, dummyActions)
+      ).to.be.revertedWith(
+        customError(
+          'DaoUnauthorized',
+          dao.address,
+          plugin.address,
+          plugin.address,
+          ownerAddress,
+          EXECUTE_PROPOSAL_PERMISSION_ID
+        )
+      );
+    });
+
+    it('correctly emits the ProposalCreated event', async () => {
+      const currentExpectedProposalId = 0;
+
+      const tx = await plugin.executeProposal(dummyMetadata, dummyActions);
+
+      await expect(
+        await plugin.executeProposal(dummyMetadata, dummyActions)
+      ).to.emit(plugin, EVENTS.ProposalCreated);
+
+      const event = await findEvent(tx, EVENTS.ProposalCreated);
+
+      expect(event.args.proposalId).to.equal(currentExpectedProposalId);
+      expect(event.args.creator).to.equal(ownerAddress);
+      expect(event.args.metadata).to.equal(dummyMetadata);
+      expect(event.args.actions.length).to.equal(1);
+      expect(event.args.actions[0].to).to.equal(dummyActions[0].to);
+      expect(event.args.actions[0].value).to.equal(dummyActions[0].value);
+      expect(event.args.actions[0].data).to.equal(dummyActions[0].data);
+    });
+
+    it('correctly emits ProposalExecuted event', async () => {
```

Comment:
```suggestion
    it('correctly emits the `ProposalExecuted` event', async () => {
```
Is it necessary to prepend/append "correctly" for the non-reverting tests? Low-importance, ignore if you prefer it like that.